activation/handler_v2_test.go

```diff
@@ -1619,20 +1631,20 @@ func Test_Marriages(t *testing.T) {
 		}
 		atx2.Sign(sig)
 		atxHandler.expectAtxV2(atx2)
-		ids := []types.NodeID{sig.NodeID(), otherSig.NodeID(), otherSig2.NodeID()}
-		for _, id := range ids {
-			atxHandler.mtortoise.EXPECT().OnMalfeasance(id)
-		}
+		atxHandler.mMalPublish.EXPECT().Publish(
+			gomock.Any(),
+			sig.NodeID(),
+			gomock.Cond(func(data any) bool {
+				proof, ok := data.(*wire.ATXProof)
+				if !ok {
+					return false
+				}
+				return proof.ProofType == wire.DoubleMarry
+			}),
+		)
 		err = atxHandler.processATX(context.Background(), "", atx2, time.Now())
 		require.NoError(t, err)
 
-		// All 3 IDs are marked as malicious
-		for _, id := range ids {
-			malicious, err := identities.IsMalicious(atxHandler.cdb, id)
-			require.NoError(t, err)
-			require.True(t, malicious)
-		}
```

Comment:
Should the `mMalPublish.EXPECT().Publish(` be extended to verify if it contains certificates for all 3 IDs (perhaps add a TODO)?

Reply:
That's not part of the ATX malfeasance publisher but rather the general publisher, it takes a proof for a certain identity from any domain and extends it with certificates to proof that this identity belongs in the same set as others.